### PR TITLE
feat: nginx rate limit configurable

### PR DESCRIPTION
**Pull request by pedrotambo**

### CODE REVIEW

File: init.sh [hunk 1]
```diff
@@ -313,8 +354,7 @@ fi
 export nginx_url="$(echo "${CATALYST_URL##*/}")"
 if [ "${CATALYST_URL}" != "http://localhost" ]; then
     echo "## Using HTTPS."
-    echo -n "## Replacing value \"\$katalyst_host\" on nginx server file ${nginx_url}... "
-    sed "s/\$katalyst_host/${nginx_url}/g" ${nginx_server_template_https} > ${nginx_server_file}
+    generate_nginx_config_files $nginx_server_template_https
```
Comment (agusaldasoro) [inline init.sh:357]: nice!

File: local/nginx/conf.d/katalyst-http.conf.template [hunk 2]
```diff
@@ -5,8 +5,8 @@ include /etc/nginx/include/rates.conf;
 
 # Maps ip address to $user_identifier variable if request is of type GET
 map $request_method $user_identifier {
-    'GET'       $http_x_forwarded_for;
-    default     '';
+    'OPTIONS'   '';
```
Comment (agusaldasoro) [inline local/nginx/conf.d/katalyst-http.conf.template:8]: need to sync with base branch

**Review by agusaldasoro — CHANGES_REQUESTED** [review]
Can we delete the files `lambdas.conf`, `content.conf`, `comms.conf` & `rates.conf` so we only have the templates on the repository, and the ones used for config are in the `.gitignore`. This way we would avoid editing the wrong file.

File: .env-advanced.example [hunk 3]
```diff
@@ -3,4 +3,6 @@ ETH_NETWORK=mainnet
 LOG_LEVEL=debug
 METRICS=true
 REGENERATE=0
-DOCKER_TAG=latest
\ No newline at end of file
+DOCKER_TAG=latest
+RATE_LIMIT_ENABLED=1
```
[the remark this answers is not in the document]
Comment (pedrotambo) [reply .env-advanced.example:7]: 👍

File: .env-advanced.example [hunk 4]
```diff
@@ -3,4 +3,6 @@ ETH_NETWORK=mainnet
...
+RATE_LIMIT_NGINX_VAR=http_x_forwarded_for
```
[the remark this answers is not in the document]
Comment (pedrotambo) [reply .env-advanced.example:8]: Yeah... i thought about that, it seemed to me a little redundant to set the config to the default values, as they are set in the `init.sh`. Wouldn't it be better to remove the from the example as they are default values?

File: README.md [hunk 5]
```diff
@@ -44,6 +44,9 @@ There is also some advanced configuration in the [.env-advanced](.env-advanced)
...
 | REGENERATE  | This will instruct the script to regenerate the certs. `0` will keep the certificates, `1` will ask for certificate renewal. If there are no certificates, the initialization script will generate them automatically, regardless of this value. For more information, look at FAQ questions (2), (3) and (4) |    0    |    no    |
+| RATE_LIMIT_ENABLED  | This will enable/disable nginx rate-limiting. `0` for false, `1` for true. |    0    |    no    |
+| RATE_LIMIT_NGINX_VAR  | This is the variable that nginx will used to limit the requests, see http://nginx.org/en/docs/http/ngx_http_core_module.html#variables for available variables. By default, the value holds a binary representation of a client’s IP address. In case that the Catalyst is behind a reverse proxy, usually some HTTP headers are set, you could use any of that headers like this `http_<HEADER_NAME>` . For example, when the widely used `x-forwarded-for` header is set, you could set `http_x-forwarded-for` |    binary_remote_addr    |    no    |
```
[the remark this answers is not in the document]
Comment (pedrotambo) [reply README.md:48]: 👍

File: .env-advanced.example (hunk 4 above)
[the remark this answers is not in the document]
Comment (agusaldasoro) [reply .env-advanced.example:8]: You are right, anyway, I'd prefer to leave it twice and with the same (default) values, as the people using this repository to run a catalyst may not have the full context of what each config is doing and they can make typos.

File: init.sh [hunk 6]
```diff
@@ -162,6 +170,35 @@ printMessage () {
...
+  # Creating content.conf, lambdas.conf and comms.conf with or without rate limiting enabled
+  if test $RATE_LIMIT_ENABLED -eq 1; then
+    cp ${nginx_content_template_file} ${nginx_content_file}
+    cp ${nginx_comms_template_file} ${nginx_comms_file}
+    cp ${nginx_lambdas_template_file} ${nginx_lambdas_file}
+  else
+    sed "/limit_req/,+1d" ${nginx_content_template_file} > ${nginx_content_file}
+    sed "/limit_req/,+1d" ${nginx_comms_template_file} > ${nginx_comms_file}
+    sed "/limit_req/,+1d" ${nginx_lambdas_template_file} > ${nginx_lambdas_file}
+  fi
```
[the remark this answers is not in the document]
Comment (pedrotambo) [reply init.sh:193]: Hmmm don't think so. `sed` with  `>` op directly writes to the `$nginx_content_file`, so leaving the `cp` commands outside the `else` will override the `sed` commands.

File: init.sh [hunk 7]
```diff
@@ -162,6 +170,35 @@ printMessage () {
...
+  printMessage ok
+  echo "## Generated nginx config files:"
+  echo -n " - $nginx_server_file, from template: " ; echo -e "\033[33m ${1} \033[39m"
```
[the remark this answers is not in the document]
Comment (pedrotambo) [reply init.sh:196]: The name of the template file, i.e., `local/nginx/include/comms.conf.template`.

File: init.sh [hunk 8]
```diff
@@ -162,6 +170,35 @@ printMessage () {
...
+  # Replacing $RATE_LIMIT_NGINX_VAR and $katalyst_host in $nginx_server_file
+  sed "s/\${RATE_LIMIT_NGINX_VAR}/\$${RATE_LIMIT_NGINX_VAR}/g;s/\$katalyst_host/${nginx_url}/g" $1 > ${nginx_server_file}
+  # Replacing $RATE_LIMIT_NGINX_VAR for rates.conf
+  sed "s/\${RATE_LIMIT_NGINX_VAR}/\$${RATE_LIMIT_NGINX_VAR}/g" ${nginx_rates_template_file} > ${nginx_rates_file}
+  # Creating content.conf, lambdas.conf and comms.conf with or without rate limiting enabled
+  if test $RATE_LIMIT_ENABLED -eq 1; then
+    cp ${nginx_content_template_file} ${nginx_content_file}
+    cp ${nginx_comms_template_file} ${nginx_comms_file}
+    cp ${nginx_lambdas_template_file} ${nginx_lambdas_file}
+  else
```
[the remark this answers is not in the document]
Comment (pedrotambo) [reply init.sh:189]: good catch

File: init.sh [hunk 9]
```diff
@@ -4,6 +4,14 @@ export data_path="local/certbot"
 export nginx_server_file="local/nginx/conf.d/00-katalyst.conf"
 export nginx_server_template_http="local/nginx/conf.d/katalyst-http.conf.template"
 export nginx_server_template_https="local/nginx/conf.d/katalyst-https.conf.template"
+export nginx_rates_file="local/nginx/include/rates.conf"
+export nginx_rates_template_file="local/nginx/include/rates.conf.template"
```
[the remark this answers is not in the document]
Comment (pedrotambo) [reply init.sh:8]: do you mean to use `nginx_rates_file + '.template` and remove `nginx_rates_template_file` var, right? Okay

**Review by agusaldasoro — APPROVED** [review]

**Pull request merged**